src/ray/object_manager/test/object_manager_stress_test.cc

```diff
@@ -0,0 +1,395 @@
+             std::unique_ptr<boost::asio::io_service> object_manager_service,
+             const ObjectManagerConfig &object_manager_config,
+             std::shared_ptr<gcs::AsyncGcsClient> gcs_client)
+      : object_manager_acceptor_(
+            main_service, boost::asio::ip::tcp::endpoint(boost::asio::ip::tcp::v4(), 0)),
+        object_manager_socket_(main_service),
+        gcs_client_(gcs_client),
+        object_manager_(main_service, std::move(object_manager_service),
+                        object_manager_config, gcs_client) {
+    RAY_CHECK_OK(RegisterGcs(main_service));
+    // Start listening for clients.
+    DoAcceptObjectManager();
+  }
+
+  ~MockServer() {
+    (void)gcs_client_->client_table().Disconnect();
+    RAY_CHECK_OK(object_manager_.Terminate());
+  }
+
+ private:
+  ray::Status RegisterGcs(boost::asio::io_service &io_service) {
+    RAY_RETURN_NOT_OK(gcs_client_->Connect("127.0.0.1", 6379));
+    RAY_RETURN_NOT_OK(gcs_client_->Attach(io_service));
+
+    boost::asio::ip::tcp::endpoint endpoint = object_manager_acceptor_.local_endpoint();
+    std::string ip = endpoint.address().to_string();
+    unsigned short object_manager_port = endpoint.port();
+
+    ClientTableDataT client_info = gcs_client_->client_table().GetLocalClient();
+    client_info.node_manager_address = ip;
+    client_info.node_manager_port = object_manager_port;
+    client_info.object_manager_port = object_manager_port;
+    return gcs_client_->client_table().Connect(client_info);
+  }
+
+  void DoAcceptObjectManager() {
+    object_manager_acceptor_.async_accept(
+        object_manager_socket_, boost::bind(&MockServer::HandleAcceptObjectManager, this,
+                                            boost::asio::placeholders::error));
+  }
+
+  void HandleAcceptObjectManager(const boost::system::error_code &error) {
+    ReceiverClientHandler client_handler =
+        [this](std::shared_ptr<ReceiverConnection> client) {
+          object_manager_.ProcessNewClient(client);
+        };
+    ReceiverMessageHandler message_handler =
+        [this](std::shared_ptr<ReceiverConnection> client, int64_t message_type,
+               const uint8_t *message) {
+          object_manager_.ProcessClientMessage(client, message_type, message);
+        };
+    // Accept a new local client and dispatch it to the node manager.
+    auto new_connection = ReceiverConnection::Create(client_handler, message_handler,
+                                                     std::move(object_manager_socket_));
+    DoAcceptObjectManager();
+  }
+
+  friend class StressTestObjectManager;
+
+  boost::asio::ip::tcp::acceptor object_manager_acceptor_;
+  boost::asio::ip::tcp::socket object_manager_socket_;
+  std::shared_ptr<gcs::AsyncGcsClient> gcs_client_;
+  ObjectManager object_manager_;
+};
+
+class TestObjectManagerBase : public ::testing::Test {
+ public:
+  TestObjectManagerBase() { RAY_LOG(INFO) << "TestObjectManagerBase: started."; }
```

Comment:
This log message is unnecessary (just in the interest of minimizing Travis and Jenkins logs for debugging later).